### DISCUSSION TITLE
Create a HandGrab

### DISCUSSION THREAD

**Discussion by m-stpz** [pmndrs/xr#296] [Q&A] [answered]
Hello!

I'm working on implementing a HandGrab feature to enhance the user experience in my VR environment. With the current RayGrab, I've noticed that users can grab objects without their virtual hands actually making contact with them. This discrepancy between the hand's position and the object's position diminishes the immersive experience.

Key improvements I'm aiming for:

- Precision: I want the grabbing action to be precise, meaning the user's virtual hand should be in close proximity or touching the object when they intend to grab it.
- Intuitive Interaction: Users should be able to reach inside or around an object to grab it, rather than just from its surface.

Could you guide me on how to implement these enhancements? 

Thank you!

**Comment by brownie11636** [comment]
why don't you use object3d.attach() like this example(https://threejs.org/examples/?q=vr#webxr_vr_handinput_pointerdrag)?

**Comment by bbohlender** [comment, marked as answer]
with react-three/xr v6, grabbing is as simple as using pointer-events with the mouse using pointer-capture on html/css.
As a side node, the roadmap contains the react-three/controls package, which will contain the DragControls component, which will allow to make anything draggable using rays but also grabbing and will work out of the box.
